dashboard/src/components/AppView/AppControls/AppControls.test.tsx

```diff
@@ -21,54 +50,55 @@ const defaultProps = {
   push: jest.fn(),
 } as IAppControlsProps;
 
-it("calls delete function when clicking the button", done => {
-  const props = { ...defaultProps, deleteApp: jest.fn().mockReturnValue(true) };
-  const wrapper = shallow(<AppControls {...props} />);
-  const button = wrapper
-    .find(".AppControls")
-    .children()
-    .find(".button-danger");
+it("calls delete function without purge when clicking the button", done => {
+  const store = mockStore(initialState);
+  const push = jest.fn();
+  const deleteApp = jest.fn().mockReturnValue(true);
+  const props = {
+    ...defaultProps,
+    deleteApp,
+    push,
+  };
+  const wrapper = getWrapper(store, props);
+  const appControls = wrapper.find(".AppControls");
+  const button = appControls.children().find(".button-danger");
   expect(button.exists()).toBe(true);
   expect(button.text()).toBe("Delete");
   button.simulate("click");
 
-  const confirm = wrapper
-    .find(".AppControls")
-    .children()
-    .find(ConfirmDialog);
+  const confirm = appControls.children().find(ConfirmDialog);
   expect(confirm.exists()).toBe(true);
   confirm.props().onConfirm(); // Simulate confirmation
 
-  expect(wrapper.state("deleting")).toBe(true);
```

Comment:
we should block this until we find a way to get the state? We are using the state in several tests, we would lose coverage. What's your plan here?

Reply:
Sorry, I forgot to include the info I found: Most people (myself included) recommend *not testing state* (ie. not testing implementation details) but instead validating what's rendered (the effect of the state. See this answer: https://stackoverflow.com/a/58432424 which I'd mostly agree with).

The quick and dirty solution on the same answer can actually be used by us (small tweak because we can't use `shallow` here, so don't have `dive()`. I had tried `appControls.state()` above which didn't work (same error), but what I didn't try, which *does* work, is explicitly `wrapper.find(AppControls).state()`. So I'm updating to that so we have a record of how we can do a quick-n-dirty change, but ultimately we should not be testing implementation details IMO.

Reply:
Generally I agree but sometimes is much faster to just test the state, is a trade-off.